modules/javafx.graphics/src/main/native-glass/mac/GlassViewDelegate.m

```diff
@@ -1045,24 +1048,25 @@ - (void)startDrag:(NSDragOperation)operation
                     offset.y = imageHalfY * (offset.y > 0 ? 1 : -1);
                 }
 
-                dragPoint.x += offset.x;
-                dragPoint.y -= offset.y;
+                //dragPoint.x += offset.x;
+                //dragPoint.y -= offset.y;
             }
         }
         else
         {
             // last resource: "empty" image
-            image = [[NSImage alloc] initWithSize:NSMakeSize(1.0f, 1.0f)];
+            rect.size.width = 1.0f;
+            rect.size.height = 1.0f;
+            image = [[NSImage alloc] initWithSize:NSMakeSize(rect.size.width, rect.size.height)];
         }
-        [self->nsView dragImage:image at:dragPoint offset:NSZeroSize event:self->lastEvent pasteboard:pasteboard source:self->nsView slideBack:YES];
 
-        // main thread blocked here until drag completes
+        rect.origin = NSPointToCGPoint(dragPoint);
+        [item setDraggingFrame:rect contents:image];
+    }
 
-        [GlassDragSource setDelegate:nil];
```

Comment:
I don't see any place where the delegate is set back to `nil` in this PR so `isDelegateSet` will always return `true` after a mouse drag. And it could turn into a dangling pointer if the GlassViewDelegate goes away.